client/my-sites/email/email-providers-stacked-comparison/index.tsx

```diff
@@ -109,16 +107,8 @@ const EmailProvidersStackedComparison = ( {
 			} )
 		);
 
-		page(
-			emailManagementPurchaseNewEmailAccount(
-				selectedSite.slug,
-				selectedDomainName,
-				currentRoute,
-				null,
-				selectedEmailProviderSlug,
-				newIntervalLength
-			)
-		);
+		const queryString = stringify( { interval: newIntervalLength } );
+		page( `${ currentRoute }?${ queryString }` );
```

Comment:
Suggestion: I think we should track `currentRoute` in the Tracks event if the switch may be triggered from different contexts.